common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -875,6 +875,23 @@ def user_details_force_sync(auth_entry, strategy, details, user=None, *args, **k
                 changed[provider_field] = current_value
                 setattr(model, field, provider_value)
 
+        # Generate fullname only for IES IDP.
+        try:
+            if (not details.get('fullname') and details.get('first_name') and details.get('last_name') and
```

Comment:
This is **very** confusing, so
1. Is there a scenario when "first_name" and "last_name" do not appear?
2. If you are trying to check if the provider is IES, please create a list as a constant and check against that list.
3. If the line is too long, do not disable the Pylint error, just split it into different lines.. @ivanvgh

Reply:
1. Yes, when the IDP is miss-configured.
2. Done.
3. Done.